### PR TITLE
faq: I get an IntegrityError or similar - what now?

### PR DESCRIPTION
**Pull request by enkore**
unmerged branch (2.3.17)

Don't remember any more whether this was already discussed.

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -186,6 +189,46 @@ Yes, if you want to detect accidental data damage (like bit rot), use the
...
+issues. We recommend to first run without ``--repair`` to assess the situation and
+if the found issues / proposed repairs sound right re-run it with ``--repair`` enabled.
+
+When errors are intermittent the cause might be bad memory, running memtest86+ or a similar
+test is recommended.
+
+A single error does not indicate bad hardware or a Borg bug -- all hardware has a certain
+bit error rate (BER), for hard drives this is typically specified as less than one error
+every 12 to 120 TB (one bit error in 10e14 to 10e15 bits) and often called
+*unrecoverable read error rate* (URE rate).
```
Comment (ThomasWaldmann) [inline docs/faq.rst]: I wouldn't put this below the memtest86+ paragraph. Usually, if memtest86+ finds a single error, you HAVE bad hardware.

File: docs/faq.rst [hunk 2]
```diff
@@ -186,6 +189,46 @@ Yes, if you want to detect accidental data damage (like bit rot), use the
...
+indicates a bad cable. If the *entire drive* is failing, then all data should be copied
+off it as soon as possible.
+
+Some drives log IO errors, which are also logged by the system (refer to the journal/dmesg).
+IO errors that impact only the filesystem can go unnoticed, since they are not reported
+to applications (e.g. Borg), but can still corrupt data.
+
+If any of these are suspicious, a self-test is recommended::
+
+    # smartctl -t long /dev/sdSomething
```
Comment (ThomasWaldmann) [inline docs/faq.rst]: Hmm, if I see the drive is bad from -a, I do not need to run the long test, I'ld rather quickly make sure I have all data backed up and get a new drive.

If the SMART data tells "all ok", then running a long test to find if that is really true for the complete media surface is interesting, errors on unused sectors could be discovered then.

**Comment by enkore** [comment]
Rewrote the section. I'm not yet _super_ happy with this, but the direction seems right.

File: docs/faq.rst [hunk 3]
```diff
@@ -186,6 +189,85 @@ Yes, if you want to detect accidental data damage (like bit rot), use the
...
+
+.. note::
+
+   Hardware diagnostics are operating system dependent and do not
+   apply universally. The commands shown apply for popular Unix-like
+   systems. Refer to your operating system's manual.
+
+Checking hard drives
+  Find the drive containing the repository and use *findmnt*, *mount* or *lsblk*
+  to learn the device path (typically */dev/...*) of the drive.
```
Comment (ThomasWaldmann) [inline docs/faq.rst]: This assumes the error was triggered by a repo access.

But errors can also get triggered by accessing the source data (in a backup run) or the target data (in an extraction run).

Comment (ThomasWaldmann) [inline docs/faq.rst]: hmm, as the whole section is about integrityerrors, my last comment does not apply.

Comment (ThomasWaldmann) [inline docs/faq.rst]: otoh, it says "an IntegrityError or similar"....

**Comment by enkore** [comment]
Yeah I see what you mean there.

I'll just merge it for now. If I (or anyone, really) gets a specific idea about improving it... PR/mail/issue.

**Pull request merged**